### PR TITLE
Fix no-missing-import for typescript

### LINKED ISSUE #23: n/no-missing-import does not recognize acceptable extensions from typescript files
Similar to n/file-extension-in-import - I'll take a look at reusing the code and PR.

### PR DESCRIPTION
**Pull request by rosskevin**
Closes #23

Refactored typescript logic into utilities for reuse.

### CODE REVIEW

File: lib/rules/file-extension-in-import.js [hunk 1]
```diff
@@ -6,15 +6,11 @@
 
 const path = require("path")
 const fs = require("fs")
+const mapTypescriptExtension = require("../util/map-typescript-extension.js")
```
Comment (aladdin-add) [inline lib/rules/file-extension-in-import.js:9]:
```suggestion
const mapTypescriptExtension = require("../util/map-typescript-extension")
```

File: lib/util/check-existence.js [hunk 2]
```diff
@@ -4,8 +4,11 @@
  */
 "use strict"
 
+const path = require("path")
 const exists = require("./exists")
 const getAllowModules = require("./get-allow-modules")
+const isTypescript = require("./is-typescript.js")
+const mapTypescriptExtension = require("../util/map-typescript-extension.js")
```
Comment (aladdin-add) [inline lib/util/check-existence.js:11]:
```suggestion
const isTypescript = require("./is-typescript")
const mapTypescriptExtension = require("../util/map-typescript-extension")
```

File: tests/lib/rules/no-missing-import.js [hunk 3]
```diff
@@ -58,6 +58,14 @@ ruleTester.run("no-missing-import", rule, {
             filename: fixture("test.js"),
             code: "import a from './a.js';",
         },
+        {
+            filename: fixture("test.ts"),
+            code: "import a from './a.js';",
+        },
+        {
+            filename: fixture("test.ts"),
+            code: "import a from './d.ts';",
```
Comment (aladdin-add) [inline tests/lib/rules/no-missing-import.js:67]: not sure I got the point - shouldn't it be `./d.js`?

Comment (rosskevin) [reply tests/lib/rules/no-missing-import.js]: yes, you are right about that. That would have failed a tsc build.

**Comment by rosskevin** [comment]
Hold, I'm trying this on my codebase and I might have some more to push.

**Pull request merged**